stan/math/prim/arr/err/is_matching_size.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef STAN_MATH_PRIM_ARR_ERR_IS_MATCHING_SIZE_HPP
+#define STAN_MATH_PRIM_ARR_ERR_IS_MATCHING_SIZE_HPP
+
+#include <stan/math/prim/scal/err/is_size_match.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * @return <code>true</code> if the variable sizes match
+ *
+ * This function checks the runtime sizes for
+ * variables that implement a <code>size()</code>
+ * method.
+ *
+ * @tparam T_y1 Type of the first variable, requires function
+ *    access to <code>.size()</code>
```

Comment:
maybe instead of "requires function access to" something like "requires class method"?